src/gui/dialogs/DialogMoveZ.cpp

```diff
@@ -0,0 +1,134 @@
+
+#include "DialogMoveZ.hpp"
+#include "ScreenHandler.hpp"
+#include "marlin_client.h"
+#include "menu_vars.h"
+#include "resource.h"
+
+DialogMoveZ::DialogMoveZ()
+    : AddSuperWindow<IDialog>(GuiDefaults::RectScreenNoFoot)
+    , value(marlin_vars()->pos[2])
+    , lastQueuedPos(marlin_vars()->pos[2])
+    , extension_width(GuiDefaults::FontBig->w * GuiDefaults::infoMaxLen)
+    , axisText(this, Rect16(), is_multiline::no, is_closed_on_click_t::no, _(axisLabel))
+    , infoText(this, Rect16(), is_multiline::yes, is_closed_on_click_t::no, _(infoTextContent))
+    , closeText(this, Rect16(), is_multiline::no, is_closed_on_click_t::no, _(closeTextContent))
+    , upText(this, Rect16(), is_multiline::no, is_closed_on_click_t::no, _(upTextContent))
+    , downText(this, Rect16(), is_multiline::no, is_closed_on_click_t::no, _(downTextContent))
+    , arrows(this, { 0, 0 }, { 0, 6, 0, 6 })
+    , numb(this, Rect16(), marlin_vars()->pos[2], "%d mm", GuiDefaults::FontBig)
+    , header(this, _(headerLabel))
+    , icon(this, Rect16(), IDR_PNG_turn_knob) {
+
+    /// using window_t 1bit flag
+    flags.close_on_click = is_closed_on_click_t::yes;
+
+    header.SetIcon(IDR_PNG_z_axis_16px);
+
+    const int item_height = GuiDefaults::FontMenuItems->h + GuiDefaults::MenuPadding.top + GuiDefaults::MenuPadding.bottom;
+    padding_ui8_t padding({ 6, 0, 6, 0 });
+    //  info text
+    Rect16 infoText_rc(Left(), item_height * 1.5, Width(), 2 * item_height);
+    infoText.SetRect(infoText_rc);
+    infoText.SetPadding(padding);
+    infoText.font = GuiDefaults::FontMenuSpecial;
+    infoText.SetAlignment(Align_t::Center());
+    //  close text
+    Rect16 closeText_rc(infoText_rc, ShiftDir_t::Bottom, item_height * 1.5);
+    closeText.SetRect(closeText_rc);
+    closeText.SetPadding(padding);
+    closeText.font = GuiDefaults::FontMenuSpecial;
+    closeText.SetAlignment(Align_t::Center());
+    // icon
+    Rect16 icon_rc(Width() / 2 - 40, (2 * Height() / 3) - 27, 81, 55);
```

Comment:
This constructor is very long and it contains complex calculations combing relative positioning with magic numbers.
Please define used rectangles like this with magic numbers: 
```
static constexpr Rect16 axisTextRect = Rect16(x, ,x ,x ,x);
static constexpr Rect16 infoTextRect = Rect16(x, ,x ,x ,x);
...etc..
```
Then assign them directly in the initializer list. It will save us time in the future.